Correction of Enclosure Based Solar Output Variables

RKStrand opened the pull request:
Pull request overview
---------------------
 - Fixes #10552
 - A series of 10 variables were misnamed as "Zone" variables when in fact they should have been labeled as "Enclosure" variables.  This led to confusion for users who thought that specifying a zone name for the output variable declarations would produce output.  In fact, when the enclosure and zone name are not the same, the output would not be produced.  This corrects the problem.  Solution includes code modification for the output variable names correction, updating numerous output files in the test suite to use the correct name (idf and rvi), various transition files/code, and documentation updates.  No unit test (by permission) since this did not change executable code.  Differences in the output are all due to the variable name change.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [X] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [X] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [NO] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

RKStrand committed:
Variable Renames for Defect 10552

Renamed variables required to correct the issue with Defect #10552
Includes modified source code including table reports and modified input files in the testfiles folder.

RKStrand committed:
Doc mods for Defect 10552

Modification of the documentation for the variable name changes.  Also found two variables that were not previously shown in the list.  Plus added some text about where to find the enclosure name if the user is uncertain what it is.

RKStrand committed:
Transition Updates for Defect 10552

Modifications to the transition program and the transition report variables file for the name changes associated with this correction.

RKStrand committed:
Merge branch 'develop' into 10552EnclosureOutputVariableNameCorrection

RKStrand committed:
Fix of transition program error

It was not noticed that the energy management sensor was already showing up in the CASE list, leading to a duplicate CASE which the compiler did not appreciate.

RKStrand committed:
Merge branch 'develop' into 10552EnclosureOutputVariableNameCorrection

RKStrand committed:
Final Edits for 10552

Missed a couple of IDFs that had variables that were changed.

mjwitte commented on src/Transition/CreateNewIDFUsingRulesV24_2_0.f90:
```diff
@@ -500,6 +500,46 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
                   OutArgs(1)='*'
                   nodiff=.false.
                 ENDIF
+                IF (OutArgs(3) == 'ZONE WINDOWS TOTAL TRANSMITTED SOLAR RADIATION RATE') THEN
```
These lines shouldn't be necessary. Transition should read Report Variables 24-1-0 to 24-2-0.csv and convert everything on the list (for both output variables and EMS sensors).

RKStrand replied:
Got it--my bad.  I have backed this out in the latest commit that I just made.

mjwitte commented on src/Transition/SupportFiles/Report Variables 24-1-0 to 24-2-0.csv:
```diff
@@ -1,3 +1,13 @@
-0,0,These numbers should be the number of report variables in the following list (including deletes). Two columns/numbers.
-
+24.1.0,24.2.0,Transition notes - some of these are EMS variable names
+0,0,These numbers should be the number of report variables in the following list (including deletes). Two columns/numbers.
```
Should be `10,10,These numbers.....`.

RKStrand replied:
Also fixed in the latest commit that I just made.

mjwitte commented on testfiles/_SmallOffice_Dulles.idf:
```diff
@@ -3570,13 +3570,13 @@
   Output:Variable,*,Surface Window Transmitted Solar Radiation Rate,annual;
 
-  Output:Variable,*,Zone Exterior Windows Total Transmitted Beam Solar Radiation Rate,monthly;
+  Output:Variable,*, Enclosure Exterior Windows Total Transmitted Beam Solar Radiation Rate,monthly;
```
I'm surprised at how many test files had some of these output variables in them.

mjwitte commented on testfiles/_SmallOffice_Dulles.idf:
```diff
@@ -5057,9 +5057,9 @@
     SumOrAverage,            !- Aggregation Type for Variable or Meter 2
     Zone Windows Total Transmitted Solar Radiation Rate,  !- Variable or Meter 3 Name
```
Missed some conversions here (AUD diffs were the hint something was up). There are also AUD diffs in 5ZoneEconomicsTariffAndLifeCycleCosts, 5ZoneTDV, and several other files.

@Myoldmopar When I run this file locally, I see err diffs. There's a new warning with this branch, but no err diffs on ci:
```
   ** Warning ** Processing Monthly Tabular Reports: Variable names not valid for this simulation
   **   ~~~   ** ...use Output:Diagnostics,DisplayExtraWarnings; to show more details on individual variables.
```
And, duh, just realized, that's because locally it ran a RunPeriod simulation, but CI didn't, so these monthly reports didn't process.

RKStrand replied:
Huh, weird.  Ok, I see that somehow I missed some of these IDF files.  Not sure how, unless these were modified during recent commits.  But it is strange that it didn't show up in the CI.  Anyway, I'm working my way through this again.  Should have another commit to fix these issues today.

RKStrand replied:
Ok, just made another commit to address this.  Should be all of them now.  Thanks!

RKStrand committed:
Corrections per Review for PR10563

Backed out changes that were not need in the transition program due to the variables being listed in the .csv file.  Also made a correction to the report variables .csv file.

RKStrand committed:
Fix Name Change in More IDFs PR10563

During review, it was noticed that there were new error messages in various files.  Apparently, not all of the IDFs with the old variable names were converted to the new names.  This should now be corrected.

RKStrand committed:
More missed variables PR10563

Not sure how these keep getting missed.  Probably user error...

mjwitte committed:
Typos

mjwitte committed:
Unrelated remove bad output variables

mjwitte reviewed (approved):
Tested transition - works well now. Found a couple of typos and cleaned up some unrelated warnings in another idf. Changes pushed. Once CI comes back clean, will merge.

"just change the names of some output variables" Such a simple fix - not.
Thanks @RKStrand

mjwitte commented on testfiles/1ZoneUncontrolled_win_2.idf:
```diff
@@ -411,18 +411,6 @@
 
   Output:Variable,ZN001:WALL001:WIN001,Surface Outside Face Temperature,timestep;
 
-! Multiplier 2 causes a new window definition.
-
-  Output:Variable,ZN001:WALL001:WIN001:2,Surface Outside Face Incident Solar Radiation Rate per Area,timestep;
```
Just for the record. Window multipliers do not create new windows, they just adjust the areas. Delete these output variable objects that trip warnings .
`** Warning ** The following Report Variables were requested but not generated -- check.rdd file`
